Build script fixed: includes separated from symbols.

npatsakula opened the pull request:
Hello! I found little bug in `build.rs` and would like to help :D

## Problem

NixOS distributive (and some more) separating `libtorch` shared symbols and headers. And it would be great if we add ability to separate ENV variables for them.

## Solution

- [x] Add ENV variable (keep reverse compatibility). 
- [x] Add documentation.
- [x] Make extra test.

LaurentMazare commented on torch-sys/build.rs:
```diff
@@ -168,8 +169,8 @@ fn make<P: AsRef<Path>>(libtorch: P, use_cuda: bool, use_hip: bool) {
                 .warnings(false)
-                .include(libtorch.as_ref().join("include"))
-                .include(libtorch.as_ref().join("include/torch/csrc/api/include"))
+                .include(&includes)
```
I don't feel that this would work in the standard setting as we need the `include` subdirectory of `libtorch` but maybe I'm missing something here?

npatsakula replied:
Thank you for reply.

Yes, you're right, I'll fix it in a moment.

npatsakula commented:
@LaurentMazare, hello! Can i ask for review?

LaurentMazare commented:
Looks reasonable, could you rebase this branch? Hopefully this should get rid of the clippy error in the CI.

npatsakula commented:
> Looks reasonable, could you rebase this branch? Hopefully this should get rid of the clippy error in the CI.

Hello @LaurentMazare! I rebased my branch on new `main`, added small documentation paragraph into `README.md` and performed small improvements in `build.rs` code.

Now I think this MR is ready to merge.

LaurentMazare commented:
Merged, thanks for the PR.
